mbq/client/contrib/permissions.py

```diff
@@ -118,38 +120,41 @@ def fetch_permissions(
         except Exception as e:
             raise ServerError("Server error") from e
 
+    def fetch_permissions(
+        self, person_id: UUIDType, org_ref: UUIDType
+    ) -> FetchedPermissionsDoc:
+        logger.debug(f"Fetching permissions from OS Core: {person_id}, {org_ref}")
+
+        return self._make_get_request(
+            f"/api/v1/people/{person_id}/permissions/by-org-ref",
+            params={"org_ref": org_ref},
+        )
+
     def fetch_permissions_for_location(
         self, person_id: UUIDType, location_id: int, location_type: RefType
     ) -> FetchedPermissionsDoc:
         logger.debug(
             f"Fetching permissions from OS Core: {person_id}, {location_type} {location_id}"
         )
 
-        try:
-            return self.client.get(
-                f"/api/v1/people/{person_id}/permissions/by-location",
-                params={"location_id": location_id, "location_type": location_type},
-            )
-        except requests.exceptions.HTTPError as e:
-            response = getattr(e, "response", None)
-            if response is not None and response.status_code // 100 == 4:
-                raise ClientError("Invalid request") from e
-            raise ServerError("Server error") from e
-        except Exception as e:
-            raise ServerError("Server error") from e
+        return self._make_get_request(
+            f"/api/v1/people/{person_id}/permissions/by-location",
+            params={"location_id": location_id, "location_type": location_type},
+        )
 
     def fetch_all_permissions(self, person_id: UUIDType) -> FetchedPermissionsDoc:
         logger.debug(f"Fetching all permissions from OS Core: {person_id}")
 
-        try:
-            return self.client.get(f"/api/v1/people/{person_id}/permissions/all")
-        except requests.exceptions.HTTPError as e:
-            response = getattr(e, "response", None)
-            if response is not None and response.status_code // 100 == 4:
-                raise ClientError("Invalid request") from e
-            raise ServerError("Server error") from e
-        except Exception as e:
-            raise ServerError("Server error") from e
+        return self._make_get_request(f"/api/v1/people/{person_id}/permissions/all")
+
+    def fetch_staff_permissions(self, person_id: UUIDType) -> StaffPermissionsDoc:
+        logger.debug(f"Fetching staff permissions from OS Core: {person_id}")
+
+        data = self._make_get_request(f"/api/v1/people/{person_id}/internal-user-permissions")
+        return StaffPermissionsDoc(
```

Comment:
This is a little different than the other methods, since this is not just a simple type. I am wrapping the data from the response into a `dataclass` for typing purposes.

Reply:
are you doing this differently because you don't want to change the format of what's returned from GET `.../internal-user-permissions`?

Reply:
so this one is a `dataclass` rather than just a type def. I wanted to be explicit with what this returns just like you did with `ConvenientOrgRefs`

Reply:
I see now, I'm on board